bokeh/_templates/notebook_load.html

```diff
@@ -27,6 +27,13 @@
     {{ bokeh_js }}
     {{ bokeh_css }}
 
+    <script type="text/javascript">
+        var comm_manager=Jupyter.notebook.kernel.comm_manager
+        comm_manager.register_target('bokeh_notebook_comms', function(comm, msg){
+            comm.on_msg(Bokeh.embed.handle_notebook_comms)
+        })
```

Comment:
this seems like it needs some sort of ID that's like a "notebook session" (identifying the originating State/output_notebook). Maybe it goes in the target name ... then that's what we could use instead of doc id perhaps. In embed.coffee's "render items" where we can have a "session id" now we could alternatively have a "notebook id" which would be used to register a Jupyter comm target? not sure. The general idea is to make a notebook comm session analogous to our websocket session as much as possible and use similar codepaths, though perhaps without full syncing.

In any case ideally I think this block of code would be just a call to Bokeh.embed with less logic inline. As it is maybe Bokeh.embed.init_notebook, though with the "notebookid in render items" idea, maybe this script tag would disappear anyhow.